src/core/Header/Header.tsx

```diff
@@ -0,0 +1,163 @@
+   * (expects `HeaderBreadcrumbs`)
+   * @example
+   * <HeaderBreadcrumbs items={[
+   *   <HeaderButton key='project' name='Project A' />
+   *   <HeaderButton key='imodel' name='IModel X' />
+   * ]} />
+   */
+  breadcrumbs?: React.ReactNode;
+  /**
+   * Content displayed to the left of the user icon
+   * (expects array of `IconButton`, potentially wrapped in a `DropdownMenu`)
+   * @example
+   * [
+   *  <IconButton><SvgNotification /></IconButton>,
+   *  <DropdownMenu>
+   *   <IconButton>
+   *    <SvgHelpCircularHollow />
+   *   </IconButton>
+   *  </DropdownMenu>
+   * ]
+   */
+  actions?: React.ReactNode[];
+  /**
+   * Children is put in the center of the Header.
+   */
+  children?: React.ReactNode;
+  /**
+   * User icon
+   * It's size and transition will be handled between slim/not slim state of the
+   * Header
+   * (expects `UserIcon`, can be wrapped in `IconButton` and `DropdownMenu` if needed)
+   * @example
+   * <DropdownMenu menuItems={...}>
+   *   <IconButton styleType='borderless'>
+   *     <UserIcon ... />
+   *   </IconButton>
+   * </DropdownMenu>
+   */
+  userIcon?: React.ReactNode;
+  /**
+   * Items in the more dropdown menu.
+   * Pass a function that takes the `close` argument (to close the menu),
+   * and returns a list of `MenuItem` components.
+   */
+  menuItems?: (close: () => void) => JSX.Element[];
+  /**
+   * If true, the header height is reduced, typically used when viewing 3D content.
+   * @default false
+   */
+  isSlim?: boolean;
+  /**
+   * Provide localized strings.
+   */
+  translatedStrings?: HeaderTranslations;
+} & Omit<CommonProps, 'title'>;
+
+const defaultTranslations: HeaderTranslations = {
+  moreOptions: 'More options',
+};
+
+/**
+ * Application header
+ * @example
+ * <Header
+ *  appLogo={<HeaderLogo logo={<SvgImodelHollow />}>iTwin Application</HeaderLogo>}
+ *  breadcrumbs={
+ *   <HeaderBreadcrumbs items={[
+ *    <HeaderButton key='project' name='Project A' />
+ *    <HeaderButton key='imodel' name='IModel X' />
+ *   ]} />
+ *  }
+ *  actions={[
+ *   <IconButton><SvgNotification /></IconButton>,
+ *   <DropdownMenu>
+ *    <IconButton>
+ *     <SvgHelpCircularHollow />
+ *    </IconButton>
+ *   </DropdownMenu>
+ *  ]}
+ *  userIcon={
+ *   <DropdownMenu menuItems={...}>
+ *    <IconButton styleType='borderless'>
+ *     <UserIcon ... />
+ *    </IconButton>
+ *   </DropdownMenu>
+ *  }
+ *  isSlim
+ * />
+ */
+export const Header = (props: HeaderProps) => {
+  const {
+    appLogo,
+    breadcrumbs,
+    isSlim = false,
+    actions,
+    userIcon,
+    menuItems,
+    translatedStrings,
+    className,
+    children,
+    ...rest
+  } = props;
+  useTheme();
+  const headerTranslations = { ...defaultTranslations, ...translatedStrings };
+  return (
+    <header
+      className={cx('iui-header', { 'iui-slim': isSlim }, className)}
+      {...rest}
+    >
+      <div className='iui-left'>
+        {appLogo}
+        {breadcrumbs && <div className='iui-divider' />}
```

Comment:
Shouldnt we have divider always after app logo even if there are no breadcrumbs??
@FlyersPh9

Reply:
Yep, it's how it was, but it was a change requested yesterday by @mayank99